pkg/cmd/cli/secrets/source.go

```
@@ -0,0 +1,198 @@
+	cmd.Flags().StringVar(&o.PrivateKeyPath, "ssh-privatekey", "", "path to the ssh private key")
+	cmd.Flags().StringVar(&o.CertificatePath, "cert-path", "", "path to the certificate file")
+	cmd.Flags().StringVar(&o.GitConfigPath, "gitconfig-path", "", "path to the .gitconfig file")
+	cmd.Flags().BoolVarP(&o.PromptForPassword, "prompt", "", false, "prompt for password or token")
+	kcmdutil.AddPrinterFlags(cmd)
+
+	return cmd
+}
+
+func (o *CreateSourceSecretOptions) CreateSourceSecret() error {
+	secret, err := o.MakeSourceSecret()
+	if err != nil {
+		return err
+	}
+
+	if _, err := o.SecretsInterface.Create(secret); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (o *CreateSourceSecretOptions) MakeSourceSecret() (*api.Secret, error) {
+	secret := &api.Secret{}
+	secret.Namespace = o.SecretNamespace
+	secret.Name = o.SecretName
+	secret.Type = SecretTypeSource
+	secret.Data = map[string][]byte{}
+
+	if len(o.Username) != 0 {
+		secret.Data["username"] = []byte(o.Username)
+	}
+
+	if len(o.Password) != 0 {
+		secret.Data["password"] = []byte(o.Password)
+	}
+
+	if len(o.CertificatePath) != 0 {
+		caContent, err := ioutil.ReadFile(o.CertificatePath)
+		if err != nil {
+			return nil, err
+		}
+		secret.Data["ca.crt"] = caContent
+	}
+
+	if len(o.GitConfigPath) != 0 {
+		gitConfig, err := ioutil.ReadFile(o.GitConfigPath)
+		if err != nil {
+			return nil, err
+		}
+		secret.Data["gitconfig"] = gitConfig
+	}
+
+	if len(o.PrivateKeyPath) != 0 {
+		privateKeyContent, err := ioutil.ReadFile(o.PrivateKeyPath)
+		if err != nil {
+			return nil, err
+		}
+		secret.Data["ssh-privatekey"] = privateKeyContent
+	}
+
+	return secret, nil
+}
+
+func (o *CreateSourceSecretOptions) Complete(f *kcmdutil.Factory, args []string) error {
+	if o.PromptForPassword {
+		if cmdutil.IsTerminal(o.Reader) {
+			o.Password = cmdutil.PromptForPasswordString(o.Reader, o.Out, "Password: ")
+			if len(o.Password) == 0 {
+				return errors.New("password must be provided")
+			}
+		}
```

if they ask to prompt and don't have a terminal, should we readline anyway (so they can pipe stdin) or err? we shouldn't silently do nothing